Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -201,130 +207,130 @@ struct SymbolGraphLoader {
-        // are not marked as unavailable,
-        // and the corresponding inheritance platform has a SGF (has been registered).
-        let missingFallbackPlatforms = DefaultAvailability.fallbackPlatforms.filter {
-            !registeredPlatforms.contains($0.key) &&
-            !unconditionallyUnavailablePlatformNames.contains($0.key) &&
-            registeredPlatforms.contains($0.value)
-        }
-        // Platforms that are defined in the Info.plist that had no corresponding SGF
-        // and are not being added as fallback of another platform.
-        let missingAvailabilities = defaultAvailabilities.filter {
-            !missingFallbackPlatforms.keys.contains($0.platformName) &&
-            !registeredPlatforms.contains($0.platformName)
-        }
-        
-        unifiedGraph.symbols.values.forEach { symbol in
+        unifiedGraph.symbols.forEach { (symbolID, symbol) in
+            
+            // The fallback platforms that are not marked as unavailable in the default availability.
+            let fallbackPlatforms = DefaultAvailability.fallbackPlatforms.filter {
+                !unconditionallyUnavailablePlatformNames.contains($0.key)
+            }
+            
+            // The symbol availability grouped by interface language.
+            // Is needed to separate the availability in language sets because the symbol availability
+            // can difer betwen language variants.
+            let symbolAvailabilityByInterfaceLanguage = Dictionary(grouping: symbol.availability, by: { $0.key.interfaceLanguage })
+            
             for (selector, _) in symbol.mixins {
-                if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
-                    guard !symbolAvailability.availability.isEmpty else { continue }
-                    // For platforms with a fallback option (e.g., Catalyst and iOS), apply the explicit availability annotation of the fallback platform when it is not explicitly available on the primary platform.
-                    DefaultAvailability.fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
+                // Platforms available for the given symbol in the given language variant.
+                var platformsAvailableAtThisSymbol = symbolAvailabilityByInterfaceLanguage[selector.interfaceLanguage].flatMap {
+                    $0.flatMap { symbolAvailability in
+                        symbolAvailability.value.filter({ item in
+                            item.introducedVersion != nil
+                        }).map(\.domain?.rawValue)
+                    }
+                } ?? []
+                
+                // The symbol availability for the given selector.
+                var symbolAvailability = (symbol.mixins[selector]?[SymbolGraph.Symbol.Availability.mixinKey] as? SymbolGraph.Symbol.Availability) ?? SymbolGraph.Symbol.Availability(availability: [])
```

Comment:
FYI: If you define this as a list of availability items (by accessing the `availability` property of `Availability` then the rest of this function doesn't repeatedly need to access `symbolAvailability.availability`. Then at the end you could create a new `Availability` from the modified items.
```suggestion
                var availabilityItems = mixins.getValueIfPresent(for: SymbolGraph.Symbol.Availability.self)?.availability ?? []
```